Update the workflow sections

adbridge opened the pull request:
This commit updates the pull request sections to detail the new
pull request template.

adbridge committed:
Update the workflow sections

This commit updates the pull request sections to detail the new
pull request template.

0xc0170 commented on docs/reference/contributing/guidelines/workflow.md:
```diff
@@ -61,90 +61,142 @@ If commits do not follow the above guidelines, we may request that you modify th
+
+We initially implement new features on separate branches in the Mbed OS repository. Mbed OS maintainers or 
+tech leads may create the new branches by following the naming convention: "feature-" prefix.
+Each 
```
Each 2x?

adbridge replied:
Well spotted!

0xc0170 commented on docs/reference/contributing/guidelines/workflow.md:
```diff
@@ -61,90 +61,142 @@ If commits do not follow the above guidelines, we may request that you modify th
+
+This can be any change in the functionality, including adding a new feature, a new method or a function. Software language does not matter.
+
+A feature contribution contains a new API, capability or behavior. It does not break backward compatibility with existing APIs, capabilities or behaviors. New feature contributions are very welcome in Mbed OS. However, because they add capability to the codebase, a new feature may introduce bugs and a support burden. New features should also come with documentation, support for most targets and c ...
```
should this contain also this: "The Mbed OS technical lead must approve feature change pull requests.", this is currently what we do

adbridge replied:
Yes we can add that.

0xc0170 commented:
@AnotherButler is there chance to get this in by end of this week ? No dependency here, the template was integrated in Mbed OS already to start using simplified PR type.

adbridge committed:
Review comments addressed

Removed duplicate instance of 'Each'.
Added extra clarification that tech leads are responsible for
approving all feature PRs.

0xc0170 commented on docs/reference/contributing/guidelines/workflow.md:
```diff
@@ -61,90 +61,143 @@ If commits do not follow the above guidelines, we may request that you modify th
+##### Refactor
 
-Release: patch
+A refactor is a contribution that modifies the codebase without fixing a bug or changing the existing behavior. Examples of this are moving functions or variables between translation units, renaming source files or folders, scope modification for nonpublic code, documentation structure changes and test organization changes. There is always the risk that someone depended on the location or name be ...
```
> Refactors are candidates for feature releases.

this one should be updated to reflect PR type above - patch release

adbridge replied:
Ah just need to remove that last sentence completely as it is already under the Patch section

0xc0170 reviewed (changes requested):
One change request, otherwise LGTM

adbridge committed:
Remove superfluous sentences

AnotherButler commented:
@0xc0170 After I've finished the priority, planned work, I'll see what I can do.

0xc0170 commented:
Any update for this one?

Commit:
Edit workflow.md

Edit file, mostly for consistent capitalization, consistent person and spelling.

Commit:
Edit workflow.md

Edit file, mostly for active voice, consistent person and correct grammar.

AnotherButler commented on docs/reference/contributing/guidelines/workflow.md:
```diff
@@ -61,90 +61,131 @@ If commits do not follow the above guidelines, we may request that you modify th
 [How We Release Arm Mbed OS](../introduction/how-we-release-arm-mbed-os.html) explains Mbed OS versioning.
 
-## Pull request types
+## Pull request template
```
The blog post says the template now "includes a note about reporting and questions and enhancements to the Forum," but I don't see that here.

adbridge replied:
Ah I think the blog post might have been a little confusing. When someone raises an issue that is not a bug then the git2jira bot closes the issue with a comment stating that questions or enhancements should be raised in the forums. It has nothing to do with PRs

adbridge replied:
In the issue template:
(which is not affected by this PR)
************************************** WARNING **************************************

   The ciarcom bot parses this header automatically. Any deviation from the 
   template may cause the bot to automatically correct this header or may result in a 
   warning message, requesting updates.

   Please ensure all sections of the template below are filled in and no changes 
   are made to the template format. Only bugs should be raised here as issues. 
   Questions or enhancements should instead be raised on our forums:
   https://forums.mbed.com/ .

   *************************************************************************************

AnotherButler replied:
Thanks for clarifying 👍 
Should we mention somewhere (not in the PR section) that nonbug issues will be closed?

0xc0170 replied:
> You can submit Mbed OS bugs on the forums or directly on GitHub.

I'll send separate PR to change this in this workflow file

0xc0170 replied:
Here https://github.com/ARMmbed/mbed-os-5-docs/pull/1161

AnotherButler reviewed (approved):
Thanks for the PRs, y'all 👍

AnotherButler commented:
Could you please fix the merge conflicts, so I can merge?

adbridge committed:
Merge branch 'adbridge-development' into development

adbridge commented:
@AnotherButler conflicts sorted :)

AnotherButler commented:
Thanks, @adbridge 👍
